packages/venia-ui/lib/RootComponents/CMS/cms.js

```diff
@@ -27,12 +30,15 @@ const CMSPage = props => {
         return <div>Page Fetch Error</div>;
     }
 
-    if (loading) {
+    if (!data) {
         return fullPageLoadingIndicator;
     }
 
-    if (!data) {
-        return null;
+    // Ensure we mark the page as loading while we check the network for updates
+    if (loading && data && !isPageLoading) {
+        setPageLoading(true);
+    } else if (!loading && data && isPageLoading) {
+        setPageLoading(false);
```

Comment:
These truthy `data` checks are redundant now.

```suggestion
    if (loading && !isPageLoading) {
        setPageLoading(true);
    } else if (!loading && isPageLoading) {
        setPageLoading(false);
```

Reply:
Are we opposed to bitwise operators? I don't see any usage in the code base, but this logic piqued my interest, and I wanted to try and simplify.

```js
if (loading ^ isPageloading) {
    setPageLoading(loading);
}
```

Reply:
@tjwiebell that's really cool, and would certainly work, but I think it'd reduce peoples ability to easily understand the code, as it's not a common operator used in JS.

Reply:
@tjwiebell I'm not opposed to bitwise operators, as long as we're not inventing reasons to use them. It would be totally fine in this case.

That said, I certainly wouldn't ask anyone to use them. Since the code is perfectly clear already, we can leave it as written.